### PR TITLE
fix: Make claim button active when wallet is not connected

### LINKED ISSUE #8444: Make claim button active when wallet is not connected
image

- will be better to then open the wallet sidebar for the user instead of disabling the button 
- we can put a loader with a text "wallet" like we do with buy now on cards

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix

  ## Needs Design check

  - @exezbcz please review

  ## Context

  - [x] Closes #8444
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

Kapture 2023-12-07 at 13 29 25

  ## Copilot Summary
  
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 56a9db7</samp>

This pull request enhances the minting process of generative NFTs and fixes a minor style issue with the `NeoButton` component. It uses a custom hook to handle the wallet modal and the button state in `Generative.vue`, and adjusts the loading spinner alignment in `NeoButton.scss`.

  
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 56a9db7</samp>

> _We are the minters of the generative art_
> _We defy the chaos with our `useDoAfterLogin` hook_
> _We align the spinner with the `NeoButton` style_
> _We purge the code of the unnecessary parts_

### CODE REVIEW

File: libs/ui/src/components/NeoButton/NeoButton.scss [hunk 1]
```diff
@@ -164,8 +164,12 @@
       @apply mr-3 #{!important};
     }
 
+    .o-btn__wrapper {
```
Comment (roiLeo) [inline libs/ui/src/components/NeoButton/NeoButton.scss:167]: `elementsWrapperClass`

Comment (Jarsen136) [reply libs/ui/src/components/NeoButton/NeoButton.scss:167]: Sorry, could you expand more? This classname was defined from oruga.

Comment (roiLeo) [reply libs/ui/src/components/NeoButton/NeoButton.scss:167]: I think you can do something like:
```vue
:elements-wrapper-class="{ 'w-auto': loadingWithLabel }"
```
or
```vue
loading-class="w-auto"
```

Comment (Jarsen136) [reply libs/ui/src/components/NeoButton/NeoButton.scss:167]: Neither of these work on my side. As there are lots of style changes applied to the `'loading-with-label'`, we should not set separate props only for the changes of `width`. What do you think?

**Comment by yangwao** [comment]
Thanks!
pay 30 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $30 @ 6.01 USD/DOT ~ 4.992 $DOT
🧗 [16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z](https://polkadot.subscan.io/account/16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)
🔗 [0x4023795186f6eb2098c9001d5c3ffd6fc199b749e7f7442c7fecb872b211a859](https://polkadot.subscan.io/extrinsic/0x4023795186f6eb2098c9001d5c3ffd6fc199b749e7f7442c7fecb872b211a859)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**